hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/OzoneClientAdapterImpl.java

```diff
@@ -289,7 +291,7 @@ public boolean hasNextKey(String key) {
   @Override
   public Token<OzoneTokenIdentifier> getDelegationToken(String renewer)
       throws IOException {
-    if (!securityEnabled) {
+    if (!securityEnabled || renewer == null) {
```

Comment:
Seems issuing token even if renewer is not passed seems to be legit case.
 public void setRenewer(Text renewer) {
    if (renewer == null) {
      this.renewer = new Text();
    } else {
      HadoopKerberosName renewerKrbName = new HadoopKerberosName(renewer.toString());
      try {
        this.renewer = new Text(renewerKrbName.getShortName());
      } catch (IOException e) {
        throw new RuntimeException(e);
      }
    }
  }

Reply:
I don't think null renewer will be a legit case. HDFS (DistrubutedFIleSystem/DFSClient) returns null in this case, which matches the behavior implemented for Ozone.

Reply:
What if a use case don't have a requirement for renewing the token. In that case we should allow empty/null renewer.

Reply:
Only block token does not require renew, but it will not be retrieved from here. So I think we will not allow empty/null renewer like HDFS.